fixture-monkey-kotlin/src/main/kotlin/com/navercorp/fixturemonkey/kotlin/ExpressionGenerators.kt

```diff
@@ -472,10 +475,30 @@ private class KotlinProperty<V, R>(private val property: KProperty1<V, R>) : Pro
     private val fieldAnnotations: List<Annotation> =
         property.javaField?.annotations?.toList() ?: listOf()
 
-    override fun getType(): Class<*> = property.javaField!!.type
-
+    /**
+     * If backed only by a getter, {@code javaProperty} or {@code javaField} may be null and cause NPE.
+     * To avoid this, returns {@link Types.UnidentifiableType} when the type cannot be identified.
+     *
+     * @see com.navercorp.fixturemonkey.kotlin.expression.root
+     */
+    override fun getType(): Class<*> = property.javaField?.type ?: Types.UnidentifiableType::class.java
+
+    /**
+     * If backed only by a getter, {@code javaProperty} or {@code javaField} may be null and cause NPE.
+     * To avoid this, returns {@link Types.UnidentifiableType} when the type cannot be identified.
+     *
+     * @see com.navercorp.fixturemonkey.kotlin.expression.root
+     */
     override fun getAnnotatedType(): AnnotatedType =
-        property.javaField!!.annotatedType
+        property.javaField?.annotatedType ?: object : AnnotatedType {
+            override fun getType(): Type = Types.UnidentifiableType::class.java
+
+            override fun getAnnotations(): Array<Annotation> = emptyArray()
+
+            override fun <T : Annotation?> getAnnotation(annotationClass: Class<T>): T? = null
+
+            override fun getDeclaredAnnotations(): Array<Annotation> = emptyArray()
+        }
```

Comment:
When extracting a JavaClass from a KProperty, the type can only be resolved if it is field-backed.
If the property is backed by a getter, the actual Java type cannot be determined, which may lead to a NullPointerException.

Previously, a generic Object type was returned in such cases, but this was ambiguous.

To make the intent explicit and allow the caller to handle branching logic more clearly, the return value was changed to the existing UnidentifiableType.

Reply:
[This](https://github.com/naver/fixture-monkey/blob/main/fixture-monkey-kotlin/src/main/kotlin/com/navercorp/fixturemonkey/kotlin/expression/Expressions.kt) triggered an NPE